### ISSUE TITLE
1.7 issue NameError: name 'config' is not defined

### ISSUE THREAD

**Issue by bufke** [jezdez/django-constance#88]
Investigating now. If I break where it errors I'm able to run this to make it work

```
from django.utils.functional import SimpleLazyObject
from constance import Config
config = SimpleLazyObject(Config)
```

Here is the stacktrace
```
Traceback (most recent call last):
File "manage.py", line 10, in <module>
  execute_from_command_line(sys.argv)
File "/usr/local/lib/python2.7/dist-packages/django/core/management/__init__.py", line 385, in execute_from_command_line
  utility.execute()
File "/usr/local/lib/python2.7/dist-packages/django/core/management/__init__.py", line 354, in execute
  django.setup()
File "/usr/local/lib/python2.7/dist-packages/django/__init__.py", line 21, in setup
  apps.populate(settings.INSTALLED_APPS)
File "/usr/local/lib/python2.7/dist-packages/django/apps/registry.py", line 108, in populate
  app_config.import_models(all_models)
File "/usr/local/lib/python2.7/dist-packages/django/apps/config.py", line 202, in import_models
  self.models_module = import_module(models_module_name)
File "/usr/lib/python2.7/importlib/__init__.py", line 37, in import_module
  __import__(name)
File "/code/ecwsp/grades/models.py", line 6, in <module>
  from ecwsp.sis.models import Student, GradeScaleRule
File "/code/ecwsp/sis/models.py", line 10, in <module>
  from constance import config
ImportError: cannot import name config
```

**Comment by jezdez** [comment]
Hmm, can you check if `'constance'` is before the `grades` app in `INSTALLED_APPS`? On 1.7 it should assign the config instance to the module only via the AppConfig, so that when constance is loaded earlier the config object should be there.

**Comment by bufke** [comment]
Yep tried that first thing.  INSTALLED_APPS is `['constance', 'constance.backends.database', ......`
Still investigating.

**Comment by bufke** [comment]
Doesn't seem right but this makes it run.

```
try:                                                                             
    from django.apps import AppConfig  # noqa                                    
except ImportError:                                                              
    config = SimpleLazyObject(Config)                                            
else:                                                                            
    default_app_config = 'constance.apps.ConstanceConfig'                        
    config = SimpleLazyObject(Config)
```

As I understand the [docs](https://docs.djangoproject.com/en/1.7/ref/applications/#how-applications-are-loaded)

1. First Django imports each item in INSTALLED_APPS. At this point the constance `__init__.py` runs
2. Then Django attempts to import the models submodule of each application, if there is one. At this point models.py runs and imports constance. It errors!
3. Then Django attempts to import the models submodule of each application, if there is one. Now constance would run - but it errored in step 2.

**Comment by qdonnellan** [comment]
It's possible that this error existed before but was disguised at import time due to the file config.py existing, which has since been renamed

**Comment by jezdez** [comment]
Sigh, this really is messy. I went with a different way to make the config object lazy, does that work for you?

**Comment by bufke** [comment]
Seems to work, Thanks @jezdez Not sure what else could be done. We can't use ready() in this use case.

**Comment by jezdez** [comment]
Released as 1.0.1.
